### ISSUE TITLE
Canvas color

### ISSUE THREAD

**Issue by mbenedettini** [richardbutler/node-spritesheet#4]
Hi:

First of all, great tool, it allowed me to keep my project 100% js without requiring a sprite generation tool written in any other language.

What happened to me is that my sprites are not rectangular: they are balls with transparent background. So I had to hack the line lib/imagemagick.js:37 to replace the current white canvas with a transparent one. 

What is the rational behind using a white canvas? I think this should be configurable so that one can choose the most convenient color. I don't have time right now but if you agree with me I can submit a pull request eventually.

Thanks in advance,
Mariano.

**Comment by richardbutler** [comment]
@mbenedettini what did you hack that line to?

If I remove the canvas argument (I'm by no means an ImageMagick veteran), the command fails as it needs an image, which is provided by the generation of a `canvas:white`. The white _should_ be negated by the `-alpha transparent` argument, which renders it at 0% alpha. For me, a bunch of pngs with drop shadows are produced correctly (with transparency) in the spritesheet png.

Which version of ImageMagick are you running?

**Comment by mbenedettini** [comment]
Sorry, forgot to mention it, I used canvas:transparent so the line ended up in:

``` javascript
return this.exec("convert -size " + width + "x" + height + " canvas:transparent -alpha transparent " + filepath, function(error, stdout, stderr) {
```

And my Imagemagick version is 6.6.9.

I'm no Imagemagick expert either nor a graphics design expert, my comment was more of a question. My source pngs have an alpha channel so they are shown transparents either in a browser or in Gimp. Perhaps their format  (png with alpha channel) have something to do.

This is hard to explain as I'm no graphics expert, but here I go: I opened the resulting spritesheet png in Gimp and its alpha channel didn't have the shapes of the source pngs, instead it was just a rectangle as large as the image size.
After tweaking Imagemagick to canvas:transparent the alpha channel took the form of the source pngs shapes.
I can submit a couple of screenshots if that helps.

Mariano.
